hummingbot/market/kucoin/kucoin_api_order_book_data_source.py

```diff
@@ -0,0 +1,282 @@
+from hummingbot.core.utils import async_ttl_cache
+from hummingbot.core.utils.async_utils import safe_gather
+from hummingbot.logger import HummingbotLogger
+from hummingbot.market.kucoin.kucoin_order_book import KucoinOrderBook
+
+KUCOIN_SYMBOLS_URL = "https://api.kucoin.com/api/v1/symbols"
+KUCOIN_TICKER_URL = "https://api.kucoin.com/api/v1/market/allTickers"
+KUCOIN_DEPTH_URL = "https://api.kucoin.com/api/v1/market/orderbook/level2_100"
+
+
+class KucoinAPIOrderBookDataSource(OrderBookTrackerDataSource):
+
+    MESSAGE_TIMEOUT = 30.0
+    PING_TIMEOUT = 10.0
+
+    _kuaobds_logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._kuaobds_logger is None:
+            cls._kuaobds_logger = logging.getLogger(__name__)
+        return cls._kuaobds_logger
+
+    def __init__(self, symbols: Optional[List[str]] = None):
+        super().__init__()
+        self._symbols: Optional[List[str]] = symbols
+
+    @classmethod
+    @async_ttl_cache(ttl=60 * 30, maxsize=1)
+    async def get_active_exchange_markets(cls) -> pd.DataFrame:
+        """
+        Returned data frame should have symbol as index and include usd volume, baseAsset and quoteAsset
+        """
+        async with aiohttp.ClientSession() as client:
+
+            market_response, exchange_response = await safe_gather(
+                client.get(KUCOIN_TICKER_URL),
+                client.get(KUCOIN_SYMBOLS_URL)
+            )
+            market_response: aiohttp.ClientResponse = market_response
+            exchange_response: aiohttp.ClientResponse = exchange_response
+
+            if market_response.status != 200:
+                raise IOError(f"Error fetching Kucoin markets information. "
+                              f"HTTP status is {market_response.status}.")
+            if exchange_response.status != 200:
+                raise IOError(f"Error fetching Kucoin exchange information. "
+                              f"HTTP status is {exchange_response.status}.")
+
+            market_data = await market_response.json()
+            exchange_data = await exchange_response.json()
+
+            attr_name_map = {"baseCurrency": "baseAsset", "quoteCurrency": "quoteAsset"}
+
+            trading_pairs: Dict[str, Any] = {
+                item["symbol"]: {attr_name_map[k]: item[k] for k in ["baseCurrency", "quoteCurrency"]}
+                for item in exchange_data["data"]
+                if item["enableTrading"] == true
+            }
+
+            market_data: List[Dict[str, Any]] = [
+                {**item, **trading_pairs[item["symbol"]]}
+                for item in market_data["data"]["ticker"]
+                if item["symbol"] in trading_pairs
+            ]
+
+            # Build the data frame.
+            all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="symbol")
+            all_markets.loc[:, "USDVolume"] = all_markets.volvalue
```

Comment:
The `volvalue` here does not necessarily mean the `USDVolume` for the particular trading pair. 
According to their API, it is just 

> The executed amount in quote currency

In cases like `LOOM-BTC`, you will be required to multiply the `volvalue` field with the `last` field(last trade price) of `BTC-USDT` to obtain the `USDVolume` for the trading pair.